command/agent/agent_endpoint_test.go

```diff
@@ -446,6 +447,218 @@ func TestHTTP_AgentMonitor(t *testing.T) {
 	})
 }
 
+func TestHTTP_AgentMonitorExport(t *testing.T) {
```

Comment:
I went to run the tests with `go test -v -count=1 -race ./command/agent -run Monitor` and found a bunch of races in the test code... but it turns out those are all in the existing `TestHTTP_AgentMonitor` and not this test :facepalm: Good topic for follow-up PR.

Reply:
Welp, I know someone who has invested a bunch of time getting familiar with our streaming logic! Can I toss a story on the minor board?